### PR TITLE
feat(yarn coverage)

### PR DESCRIPTION
**Pull request by odyslam**
- monorepo wide`yarn coverage`
- code coverage for all contracts* packages (it requires forge, lcov)
- coverage is generated in: `coverage/<package>/index.html`
- @prestwich @anna-carroll let me know if you want to commit the lcov report to the repo

cc @Imti in case coverage is added for other packages as well

### CODE REVIEW

File: scripts/coverage-report.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ $(which genhtml) == "" ]]; then
+  echo "To generate code coverage, you need to install lcov"
```
Comment (prestwich) [inline scripts/coverage-report.sh:6]: should be documented in contributing.sh

File: scripts/coverage-report.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+if [[ $(which forge) == "" ]]; then
+  echo "To generate code coverage, you need to install Foundry"
+  echo "See: https://getfoundry.sh/"
+  exit 1
+fi
+
+if [[ $1 == *"core"* ]]; then
+  # Need to be to top-level directory of the monorepo
+  cd $(git rev-parse --show-toplevel)
+  path="coverage/contracts-core"
```
Comment (prestwich) [inline scripts/coverage-report.sh:20]: should it be `coverage/` or `.coverage/` ? I think I prefer with .

**Review by prestwich — CHANGES_REQUESTED** [review]
gitignore the coverage folder

File: scripts/coverage-report.sh (hunk 1 above)
Comment (prestwich) [reply scripts/coverage-report.sh:6]: errr contributing.md

File: packages/sdk/.gitignore [hunk 3]
```diff
@@ -2,4 +2,4 @@
 dist/
 tsconfig.tsbuildinfo
 .nyc_output/
-coverage/
+.coverage/
```
Comment (prestwich) [inline packages/sdk/.gitignore:5]: this should be in the global get ignore as a wildcard. not replicated in each package

File: packages/contracts-bridge/.gitignore [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
+.coverage/
 coverage.json
 .env
 src.ts/
 src/
 dist/
-foundry_artifacts/
\ No newline at end of file
+foundry_artifacts/
+.coverage
```
Comment (prestwich) [inline packages/contracts-bridge/.gitignore:11]: duplicated from line 4

File: packages/contracts-bridge/package.json [hunk 5]
```diff
@@ -28,6 +28,7 @@
     "test:unit": "FOUNDRY_PROFILE=bridge-ci forge test",
     "test:unit:local": "FOUNDRY_PROFILE=bridge forge test",
     "test:static-analyze": "rm -rf node_modules && yarn link:node_modules && slither .",
+    "test:coverage": "cd ../.. && scripts/coverage-report.sh bridge",
```
Comment (prestwich) [inline packages/contracts-bridge/package.json:31]: can this be invoked without `cd` e.g. `../../scripts/coverage-report.sh bridge`

if not, why not

File: scripts/coverage-report.sh [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  echo "Report generated at $path"
+  echo "Open $(readlink -f $path)/index.html"
+  rm lcov.info
+elif [[ $1 == *"router"* ]]; then
+  # Need to be to top-level directory of the monorepo
+  cd $(git rev-parse --show-toplevel)
+  path="packages/contracts-router/.coverage"
+  FOUNDRY_PROFILE=router forge coverage --report lcov
+  genhtml -o "$path" lcov.info
+  echo "Report generated at $path"
```
Comment (prestwich) [inline scripts/coverage-report.sh:41]: this can be deduplicated

File: packages/contracts-router/package.json [hunk 7]
```diff
@@ -26,6 +26,7 @@
     "prepublish": "yarn bootstrap && yarn build",
     "test:unit": "FOUNDRY_PROFILE=router-ci forge test",
     "test:unit-local": "FOUNDRY_PROFILE=router forge test",
+    "test:coverage": "cd ../../scripts/coverage-report.sh router",
```
Comment (prestwich) [inline packages/contracts-router/package.json:29]: shouldn't have cd

File: packages/local-environment/jest.config.ts [hunk 8]
```diff
@@ -19,28 +19,22 @@ export default {
   clearMocks: true,
 
   // Indicates whether the coverage information should be collected while executing the test
-  collectCoverage: true,
+  // collectCoverage: true,
```
Comment (prestwich) [inline packages/local-environment/jest.config.ts:22]: should be commented? or should be removed?

**Review by prestwich — APPROVED** [review]

File: packages/local-environment/jest.config.ts (hunk 8 above)
Comment (rswanson) [reply packages/local-environment/jest.config.ts:22]: Is there a particular reason why you chose to use the --collectCoverage flag for the coverage command as opposed to using the jest.config.ts? I'm happy its being done consistently, just curious for my own knowledge why you wouldn't always want the coverage vs having a specific coverage command

Comment (odyslam) [reply packages/local-environment/jest.config.ts:22]: I have absolutely no idea, this is @Imti 
I rebased on his PR and added the contract stuff

Comment (xBalbinus) [reply packages/local-environment/jest.config.ts:22]: Noted, disabling collect coverage by default and only having coverage with flag.

**Review by xBalbinus — APPROVED** [review]

**Comment by James8T** [comment]
ETHmad holders will have recovery near 0%. Only 1-2 hackers hold the stolen ETH, probably will never been returned. 

I want you JAMES PRESTWICH and all Nomad team to contribute with own pocket, you have $ETH!!!. I cant pay scholarships to my son because of you!!! Also Why don’t you negociate with big 8 figures wallets to write-off their debt against equity?

**Pull request merged**